internal/backends/winit/event_loop.rs

```diff
@@ -643,7 +593,9 @@ pub fn run() -> Result<(), corelib::platform::PlatformError> {
 
     #[cfg(not(target_arch = "wasm32"))]
     {
-        winit_loop.run_return(
+        use winit::platform::pump_events::{EventLoopExtPumpEvents, PumpStatus};
+        while let PumpStatus::Continue = winit_loop.pump_events(
+            None,
```

Comment:
The way that I understand the pump API, this means that we pass zero as timeout and therefore this will spin the event loop at 100%. Maybe it would make more sense here to use [`run_ondemand`](https://docs.rs/winit/0.29.1-beta/winit/platform/run_ondemand/trait.EventLoopExtRunOnDemand.html#tymethod.run_ondemand) ?

Reply:
if None is used as a timeout, it blocks forever until the next event and does not spin 100% (None != Some(0))

But i'm not sure I understand the difference with run_ondemand.  I think that is exactly the same as this `while` loop i'm making.  probably run_ondemand is the right function here.

Reply:
You're right, I misread the API, so `None` is fine. I'd still go for `run_ondemand` to keep it simpler :)